docs/source/advanced/cmake.rst

```diff
@@ -1,12 +1,99 @@
+
+.. code-block:: bash
+
+   cd <alpaka_project_root>
+   cmake --list-presets
+
+To configure, build and run the tests of a specific preset, run the following commands (for the example, we use the ``cpu-serial`` preset):
+
+.. code-block:: bash
+
+   cd <alpaka_project_root>
+   # configure a specific preset
+   cmake --preset cpu-serial
+   # build the preset
+   cmake --build --preset cpu-serial
+   # run test of the preset
+   ctest --preset cpu-serial
+
+All presets are configure and build in a subfolder of the ``<alpaka_project_root>/build`` folder.
+
+Modifying and Extending Presets
+-------------------------------
+
+The easiest way to change a preset is to set CMake arguments during configuration:
+
+.. code-block:: bash
+
+   cd <alpaka_project_root>
+   # configure the cpu-serial preset with clang++ as C++ compiler
+   cmake --preset cpu-serial -DCMAKE_CXX_COMPILER=clang++
+   # build the preset
+   cmake --build --preset cpu-serial
+   # run test of the preset
+   ctest --preset cpu-serial
+
+It is also possible to configure the default setting first and then change the arguments with ``ccmake``:
+
+.. code-block:: bash
+
+   cd <alpaka_project_root>
+   # configure the cpu-serial preset with clang++ as C++ compiler
+   cmake --preset cpu-serial
+   cd build/cpu-serial
+   ccmake .
+   cd ../..
+   # build the preset
+   cmake --build --preset cpu-serial
+   # run test of the preset
+   ctest --preset cpu-serial
+
+CMake presets also offer the option of creating personal, user-specific configurations based on the predefined CMake presets. To do this, you can create the file ``CMakeUserPresets.json`` in the root directory of your project (the file is located directly next to ``CMakePresets.json``). You can then create your own configurations from the existing CMake presets. The following example takes the cpu-serial configuration, uses ``ninja`` as the generator instead of the standard generator and uses the build type ``RELEASE``.
+
+.. code-block:: json
+
+   {
+      "version": 3,
+      "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 22,
+        "patch": 0
+      },
+      "configurePresets": [
+        {
+            "name": "cpu-serial-ninja-release",
+            "inherits": "cpu-serial",
+            "generator": "Ninja",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": {
+                    "type": "STRING",
+                    "value": "RELEASE"
+                }
+            }
+        }
+      ]
+   }
+
+.. hint::
+
+   Many IDEs like `Visual Studio Code <https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md>`_ and `CLion <https://www.jetbrains.com/help/clion/cmake-presets.html>`_ supports CMake presets.
```

Comment:
```suggestion
Many IDEs like `Visual Studio Code <https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md>`_ and `CLion <https://www.jetbrains.com/help/clion/cmake-presets.html>`_ supports CMake presets.
```
Whitespace on purpose?

Also:
```suggestion
   Many IDEs like `Visual Studio Code <https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md>`_ and `CLion <https://www.jetbrains.com/help/clion/cmake-presets.html>`_ support CMake presets.
```

Reply:
The whitespaces are required by reStructuredText that it correctly renders the hint box.